api/controllers/environment.go

```diff
@@ -37,10 +41,55 @@ func (e *EnvironmentController) CreateEnvironment(request *restful.Request, resp
 }
 
 func (e *EnvironmentController) DeleteEnvironment(request *restful.Request, response *restful.Response) {
+	id := request.PathParameter("id")
+	if id == "" {
+		// todo: handle error
+	}
+
+	job := models.CreateJobRequest{
+		JobType: job.DeleteEnvironmentJob,
+		Request: id,
+	}
+
+	jobID, err := e.JobScheduler.ScheduleJob(job)
+	if err != nil {
+		// todo: handle error
+	}
+
+	WriteJobResponse(response, jobID)
 }
 
 func (e *EnvironmentController) GetEnvironment(request *restful.Request, response *restful.Response) {
+	id := request.PathParameter("id")
+	if id == "" {
+		// todo: handle error
+	}
+
+	environment := e.Provider.GetEnvironment(id)
```

Comment:
Discussions about the ambiguity of `Provider` aside, this is pretty clean from the calling code's point of view. slightly confusing to me to see an abstract object returned from `GetEnvironment` and then a model returned from just `Model`...